frontend/src/data/Dialect.ts

```diff
@@ -55,6 +55,16 @@ export default class Dialect {
     }
     return dialect;
   }
+
+  static forURI(baseURI: URI): Dialect {
```

Comment:
Ok, I see you already correct it. But still, splitting URI path is a fragile logic. In `dialects.json` there is already an association with dialect URI - it's our source of truth so it'd be better if you somehow make use of this value instead